src/Protocol/Protocol110x.cpp

```diff
@@ -0,0 +1,857 @@
+		INVULNERABLE = _FIREBALL_NEXT,
+
+		// Fireworks
+		FIREWORK_INFO = _ENTITY_NEXT,
+
+		// Hanging
+		_HANGING_NEXT = _ENTITY_NEXT,
+
+		// ItemFrame
+		ITEM_FRAME_ITEM = _HANGING_NEXT,
+		ROTATION,
+
+		// Item
+		ITEM = _ENTITY_NEXT,
+
+		// Living
+		ACTIVE_HAND = _ENTITY_NEXT,
+		HEALTH,
+		POTION_EFFECT_COLOR,
+		POTION_EFFECT_AMBIENT,
+		NUMBER_OF_ARROWS,
+		_LIVING_NEXT,
+
+		// Player
+		ADDITIONAL_HEARTHS = _LIVING_NEXT,
+		SCORE,
+		DISPLAYED_SKIN_PARTS,
+		MAIN_HAND,
+
+		// ArmorStand
+		ARMOR_STAND_STATUS = _LIVING_NEXT,
+		HEAD_ROTATION,
+		BODY_ROTATION,
+		LEFT_ARM_ROTATION,
+		RIGHT_ARM_ROTATION,
+		LEFT_LEG_ROTATION,
+		RIGHT_LEG_ROTATION,
+
+		// Insentient
+		INSENTIENT_STATUS = _LIVING_NEXT,
+		_INSENTIENT_NEXT,
+
+		// Ambient
+		_AMBIENT_NEXT = _INSENTIENT_NEXT,
+
+		// Bat
+		HANGING = _AMBIENT_NEXT,
+
+		// Creature
+		_CREATURE_NEXT = _INSENTIENT_NEXT,
+
+		// Ageable
+		BABY = _CREATURE_NEXT,
+		_AGEABLE_NEXT,
+
+		// PolarBear
+		STANDING = _AGEABLE_NEXT,
+
+		// Animal
+		_ANIMAL_NEXT = _AGEABLE_NEXT,
+
+		// Horse
+		HORSE_STATUS = _ANIMAL_NEXT,
+		HORSE_TYPE,
+		HORSE_VARIANT,
+		HORSE_OWNER,
+		ARMOR,
+
+		// Pig
+		HAS_SADDLE = _ANIMAL_NEXT,
+
+		// Rabbit
+		RABBIT_TYPE = _ANIMAL_NEXT,
+
+		// Sheep
+		SHEEP_STATUS = _ANIMAL_NEXT,
+
+		// TameableAnimal
+		TAMEABLEANIMAL_STATUS = _ANIMAL_NEXT,
+		TAMEABLE_OWNER,
+		_TAMEABLE_NEXT,
+
+		// Ocelot
+		OCELOT_TYPE = _TAMEABLE_NEXT,
+
+		// Wolf
+		WOLF_DAMAGE_TAKEN = _TAMEABLE_NEXT,
+		BEGGING,
+		COLLAR_COLOR,
+
+		// Villager
+		PROFESSION = _AGEABLE_NEXT,
+
+		// Golem
+		_GOLEM_NEXT = _CREATURE_NEXT,
+
+		// IronGolem
+		PLAYER_CREATED = _GOLEM_NEXT,
+
+		// Shulker
+		FACING_DIRECTION = _GOLEM_NEXT,
+		ATTACHMENT_POSITION,
+		SHIELD_HEIGHT,
+
+		// Monster
+		_MONSTER_NEXT = _CREATURE_NEXT,
+
+		// Blaze
+		ON_FIRE = _MONSTER_NEXT,
+
+		// Creeper
+		CREEPER_STATE = _MONSTER_NEXT,
+		CREEPER_POWERED,
+		IGNITED,
+
+		// Guardian
+		GUARDIAN_STATUS = _MONSTER_NEXT,
+		TARGET,
+
+		// Skeleton
+		SKELETON_TYPE = _MONSTER_NEXT,
+		ARMS_SWINGING,
+
+		// Spider
+		CLIMBING = _MONSTER_NEXT,
+
+		// Witch
+		AGGRESIVE = _MONSTER_NEXT,
+
+		// Wither
+		FIRST_HEAD_TARGET = _MONSTER_NEXT,
+		SECOND_HEAD_TARGET,
+		THIRD_HEAD_TARGET,
+		INVULNERABLE_TIMER,
+
+		// Zombie
+		ZOMBIE_IS_BABY = _MONSTER_NEXT,
+		ZOMBIE_TYPE,
+		CONVERTING,
+		HANDS_RISED_UP,
+
+		// Enderman
+		CARRIED_BLOCK = _MONSTER_NEXT,
+		SCREAMING,
+
+		// EnderDragon
+		DRAGON_PHASE = _INSENTIENT_NEXT,
+
+		// Flying
+		_FLYING_NEXT = _INSENTIENT_NEXT,
+
+		// Ghast
+		ATTACKING = _FLYING_NEXT,
+
+		// Slime
+		SLIME_SIZE = _INSENTIENT_NEXT,
+
+		// Minecart
+		SHAKING_POWER = _ENTITY_NEXT,
+		SHAKING_DIRECTION,
+		SHAKING_MULTIPLIER,
+		BLOCK_ID_META,
+		BLOCK_Y,
+		SHOW_BLOCK,
+		_MINECART_NEXT,
+
+		// MinecartCommandBlock
+		COMMAND = _MINECART_NEXT,
+		LAST_OUTPUT,
+
+		// MinecartFurnace
+		MINECARTFURNACE_POWERED = _MINECART_NEXT,
+
+		// TNTPrimed
+		FUSE_TIME = _ENTITY_NEXT,
+	};
```

Comment:
I agree to @Pokechu22 ! I need to copy the whole enum for the new protocol, but I only need to add the few new Metadata entries. This is the point of auto-assignment!

> Having the compiler assign numbers to enums is a sure way to introduce bugs later on when someone edits the enum without realizing this.

There are two scenarios, when somebody edits this enum:

- My enum is incorrect: Then the person fixing this knows, how the enum works, because it created the issue.

- A new Protocol is released: Then the implementing person can miss this. But there are multiple people reviewing the new protocol. Also people like @mathiascode test the new changes. The probability of a bug is rare.

And an occurring bug is easy to fix. Better having a bugged new protocol than a heavily delayed new protocol(Just look at the issue request date :smirk: ).